src/UglyToad.PdfPig/Export/HOcrTextExporter.cs

```diff
@@ -0,0 +1,358 @@
+﻿using System;
+using System.Linq;
+using UglyToad.PdfPig.Content;
+using UglyToad.PdfPig.DocumentLayoutAnalysis;
+using UglyToad.PdfPig.Geometry;
+using UglyToad.PdfPig.Util;
+
+namespace UglyToad.PdfPig.Export
+{
+    /// <summary>
+    /// hOCR v1.2 (HTML) text exporter.
+    /// <para>See http://kba.cloud/hocr-spec/1.2/ </para>
+    /// </summary>
+    internal class HOcrTextExporter : ITextExporter
+    {
+        private const string xmlHeader = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\"\n\t\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n";
+        private const string hocrjs = "<script src='https://unpkg.com/hocrjs'></script>\n";
+
+        private IPageSegmenter pageSegmenter;
+        private IWordExtractor wordExtractor;
+
+        private decimal scale;
+        private string indentChar;
+
+        private int pageCount = 0;
+        private int areaCount = 0;
+        private int lineCount = 0;
+        private int wordCount = 0;
+        private int pathCount = 0;
+        private int paraCount = 0;
+        private int imageCount = 0;
+
+        /// <summary>
+        /// hOCR v1.2 (HTML)
+        /// <para>See http://kba.cloud/hocr-spec/1.2/ </para>
+        /// </summary>
+        /// <param name="wordExtractor"></param>
+        /// <param name="pageSegmenter"></param>
+        /// <param name="scale"></param>
+        /// <param name="indent">Indent character.</param>
+        public HOcrTextExporter(IWordExtractor wordExtractor, IPageSegmenter pageSegmenter, double scale = 1.0, string indent = "\t")
+        {
+            this.wordExtractor = wordExtractor;
+            this.pageSegmenter = pageSegmenter;
+            this.scale = (decimal)scale;
+            this.indentChar = indent;
+        }
+
+        /// <summary>
+        /// Get the hORC (HTML) string of the page layout.
```

Comment:
I'd do a find and replace on `hORC <-> hOCR`, it's an easy mistake to make 😆

Reply:
hehe yes!